### ISSUE TITLE
Flutter fails to report keyboard changes via MediaQuery.of(context).viewInsets

### ISSUE THREAD

**Issue by craiglabenz** [flutter/flutter#48638]
## Steps to Reproduce

1. Load [this code](https://gist.github.com/craiglabenz/4ac72794729bf4e676f763d8a32f4511) into the simulator.
2. Tap the two inputs, pressing CMD-K (on Mac, obviously) to show the keyboard if necessary.
3. Notice the debug statements which always read 0 bottom inset.

**Target Platform:** iOS & Android
**Devices:** iPhone 11 Pro and Pixel 3

## Logs

```
flutter doctor -v
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.1 19B88, locale en-US)
    • Flutter version 1.12.13+hotfix.5 at /Users/craiglabenz/flutter
    • Framework revision 27321ebbad (5 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/craiglabenz/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[!] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554             • android-x86 • Android 10 (API 29) (emulator)
```

**Comment by danspano** [comment]
@craiglabenz I noticed you have your InputsPage widget inside of a Scaffold.
The Scaffold widget has a property "resizeToAvoidBottomInset" which defaults to true. This essentially "eats" the keyboard inset and applies padding to the child widget, then passes 0 inset in MediaQuery to the child as the padding has already been applied.

I tried running your example and setting "resizeToAvoidBottomInset" to false in the Scaffold and was able to see the padding value for the keyboard output to the console.

Try setting resizeToAvoidBottomInset to false and seeing if that resolves your issue.
